test_conformance/api/test_api_consistency.cpp

```diff
@@ -473,14 +437,689 @@ int test_consistency_pipes(cl_device_id deviceID, cl_context context,
+                                            &test_kernel, "test");
+        test_error(error, "Unable to create test kernel");
+
+        size_t sz = 0;
+
+        // clGetDeviceInfo, passing
+        // CL_DEVICE_GLOBAL_VARIABLE_PREFERRED_TOTAL_SIZE
+        // Returns 0 if device does not support Program Scope Global Variables.
+
+        error = clGetDeviceInfo(deviceID,
+                                CL_DEVICE_GLOBAL_VARIABLE_PREFERRED_TOTAL_SIZE,
+                                sizeof(sz), &sz, NULL);
+        test_error(
+            error,
+            "Unable to query CL_DEVICE_GLOBAL_VARIABLE_PREFERRED_TOTAL_SIZE");
+        test_assert_error(
+            sz == 0,
+            "CL_DEVICE_MAX_GLOBAL_VARIABLE_SIZE returned 0 but "
+            "CL_DEVICE_GLOBAL_VARIABLE_PREFERRED_TOTAL_SIZE returned a "
+            "non-zero value");
+
+        // clGetProgramBuildInfo, passing
+        // CL_PROGRAM_BUILD_GLOBAL_VARIABLE_TOTAL_SIZE
+        // Returns 0 if device does not support Program Scope Global Variables.
+
+        error = clGetProgramBuildInfo(
+            program, deviceID, CL_PROGRAM_BUILD_GLOBAL_VARIABLE_TOTAL_SIZE,
+            sizeof(sz), &sz, NULL);
+        test_error(
+            error,
+            "Unable to query CL_PROGRAM_BUILD_GLOBAL_VARIABLE_TOTAL_SIZE");
+        test_assert_error(sz == 0,
+                          "CL_DEVICE_MAX_GLOBAL_VARIABLE_SIZE returned 0 "
+                          "but CL_PROGRAM_BUILD_GLOBAL_VARIABLE_TOTAL_SIZE "
+                          "returned a non-zero value");
+    }
+
+    return TEST_PASS;
+}
+
+int test_consistency_non_uniform_work_group(cl_device_id deviceID,
+                                            cl_context context,
+                                            cl_command_queue queue,
+                                            int num_elements)
+{
+    // clGetDeviceInfo, passing CL_DEVICE_NON_UNIFORM_WORK_GROUP_SUPPORT:
+    // May return CL_FALSE, indicating that device does not support Non-Uniform
+    // Work Groups.
+    cl_int error;
+
+    const size_t allocSize = 16;
+    clMemWrapper mem;
+    clProgramWrapper program;
+    clKernelWrapper kernel;
+
+    cl_bool nonUniformWorkGroupSupport = CL_FALSE;
+    error = clGetDeviceInfo(deviceID, CL_DEVICE_NON_UNIFORM_WORK_GROUP_SUPPORT,
+                            sizeof(nonUniformWorkGroupSupport),
+                            &nonUniformWorkGroupSupport, NULL);
+    test_error(error,
+               "Unable to query CL_DEVICE_NON_UNIFORM_WORK_GROUP_SUPPORT");
+
+    if (nonUniformWorkGroupSupport == CL_FALSE)
+    {
+        // Test setup:
+
+        mem =
+            clCreateBuffer(context, CL_MEM_READ_WRITE, allocSize, NULL, &error);
+        test_error(error, "Unable to create test buffer");
+
+        error = create_single_kernel_helper_with_build_options(
+            context, &program, &kernel, 1, &test_kernel, "test",
+            "-cl-std=CL3.0");
```

Comment:
Could leave the CL C version selection to the framework. That would be a bit more future proof.

Reply:
This wasn't an option when I created this PR.  Fixed!